### PR TITLE
Fix blocks filtering - Closes #629

### LINKED ISSUE #627: Aggregated votes_received should only return non-zero aggregates
### Actual behavior
`api/v2/votes_received?aggregate=true` returns votes with vote weight `0`.

### Expected behavior
The endpoint should filter out any entries with vote weight `0` and only return `non-zero` weighted votes.

### Steps to reproduce
`http://service-v5.liskdev.net/api/v2/votes_sent?address=lskghfgmr6qnvmh9zgoeqejzosdx3tbpfopc2pcmu`

`http://service-v5.liskdev.net/api/v2/votes_received?address=lskghfgmr6qnvmh9zgoeqejzosdx3tbpfopc2pcmu&aggregate=true`

### Which version(s) does this affect? (Environment, OS, etc...)
`0.4.0`

### LINKED ISSUE #629: Filter is not working as expected in blocks monitoring page
### Expected behavior
In block monitoring page, wildcard filter (AND condition) should work when we filter with more than one criteria.

### Actual behavior
In block monitoring page, when i try to filter with multiple inputs then it is not working as expected.

### Which version(s) does this affect? (Environment, OS, etc...)
Mac, v2.0.0-rc0

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?
This PR resolves #629 

### How was it solved?
- [x] Update `isQueryFromIndex` function to query core if any one of them is available in params(id, height or heightBetween)
- [x] Added integration test cases

### How was it tested?
Local + Jenkins

### CODE REVIEW

File: services/core/shared/core/compat/sdk_v5/blocks.js [hunk 1]
```diff
@@ -300,11 +311,11 @@ const getBlocks = async params => {
 	}
 
 	try {
-		if (params.id) {
+		if (params.ids) {
+			blocks.data = await getBlocksByIDs(params.ids);
+		} else if (params.id) {
 			blocks.data = await getBlockByID(params.id);
 			if ('offset' in params && params.limit) blocks.data = blocks.data.slice(params.offset, params.offset + params.limit);
```
Comment (sameersubudhi) [inline services/core/shared/core/compat/sdk_v5/blocks.js:318]: With the updated implementation of the `isQueryFromIndex` method, the following check may not be necessary.
```suggestion
```

File: services/core/shared/core/compat/sdk_v5/blocks.js [hunk 2]
```diff
@@ -300,11 +311,11 @@ const getBlocks = async params => {
...
-		} else if (params.ids) {
-			blocks.data = await getBlocksByIDs(params.ids);
 		} else if (params.height) {
 			blocks.data = await getBlockByHeight(params.height);
 			if ('offset' in params && params.limit) blocks.data = blocks.data.slice(params.offset, params.offset + params.limit);
```
Comment (sameersubudhi) [inline services/core/shared/core/compat/sdk_v5/blocks.js:321]: Same as above.

File: services/core/methods/api_v2/controllers/blocks.js [hunk 3]
```diff
@@ -49,18 +45,6 @@ const getBlocks = async params => {
 
 		const response = await Core.getBlocks(params);
 
-		if (typeof params.blockId === 'string') {
-			if (isEmptyArray(response.data)) {
-				throw new NotFoundException(`Block ID ${params.blockId} not found.`);
```
Comment (sameersubudhi) [inline services/core/methods/api_v2/controllers/blocks.js:54]: Consider throwing `NotFoundException` at the appropriate place within `services/core/shared/core/compat/sdk_v5/blocks.js`.

File: services/core/methods/api_v2/controllers/blocks.js [hunk 4]
```diff
@@ -49,18 +45,6 @@ const getBlocks = async params => {
...
-			}
-		}
-
-		if (typeof params.height === 'string') {
-			if (isEmptyArray(response.data)) {
-				throw new NotFoundException(`Height ${params.height} not found.`);
```
Comment (sameersubudhi) [inline services/core/methods/api_v2/controllers/blocks.js:60]: Same as above.

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
Please find my observations below:

File: services/core/shared/core/compat/sdk_v5/blocks.js [hunk 5]
```diff
@@ -184,7 +189,11 @@ const getLastBlock = async () => {
 const isQueryFromIndex = params => {
 	const paramProps = Object.getOwnPropertyNames(params);
 
-	const isDirectQuery = ['id', 'height', 'heightBetween'].some(prop => paramProps.includes(prop));
+	const directQueryParams = ['id', 'height', 'heightBetween'];
+	const defaultQueryParams = ['limit', 'offset', 'sort'];
+	const isDirectQuery = directQueryParams.some(prop => paramProps.includes(prop))
+		&& (paramProps.filter(prop => directQueryParams.includes(prop))).length === 1
+		&& paramProps.every(prop => directQueryParams.concat(defaultQueryParams).includes(prop));
```
Comment (sameersubudhi) [inline services/core/shared/core/compat/sdk_v5/blocks.js:196]: Please consider commenting on what each part of the logic aims to achieve.
```suggestion
	// At least one of the 'directQueryParams' is present within the request params
	// Only one of the 'directQueryParams' is present within the request params
	// All the request params are either 'directQueryParams' or 'defaultQueryParams'
	const isDirectQuery = directQueryParams.some(prop => paramProps.includes(prop))
		&& (paramProps.filter(prop => directQueryParams.includes(prop))).length === 1
		&& paramProps.every(prop => directQueryParams.concat(defaultQueryParams).includes(prop));
```

Also, please test if the following works:
```suggestion
	// For 'isDirectQuery' to be 'true', the request params should contain exactly one of 'directQueryParams'
	// and all of them must be contained within 'directQueryParams' or 'defaultQueryParams'
	const isDirectQuery = (paramProps.filter(prop => directQueryParams.includes(prop))).length === 1
		&& paramProps.every(prop => directQueryParams.concat(defaultQueryParams).includes(prop));
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
Kindly consider the following feedback:

File: services/core/shared/core/compat/sdk_v5/blocks.js (hunk 5 above)
Comment (nagdahimanshu) [reply services/core/shared/core/compat/sdk_v5/blocks.js]: I have tested the suggestion and it worked as expected.

**Review by sameersubudhi — APPROVED** [review]
LGTM 👍

**Review by MichalTuleja — APPROVED** [review]
LGTM 👍

**Pull request merged**